Release 4.6.2

awead opened the pull request:
Adds improvements to the work edit form so that <Enter> does what is expected when submitting the form, or choosing new creators to add to the work.

cam156 commented on config/application.yml:
```diff
@@ -5,21 +5,19 @@
 # Scholarsphere::Config::REQUIREMENTS otherwise, the cap deploy will fail.
 #
 development:
-  TMPDIR: "/tmp"
```
@awead what is the reason for removing this?

awead replied:
Locally, `TMPDIR` is already defined and Figaro issues a warning. Removing this avoids the warning. Notice that the production section still has it because it's on our servers that this variable may not be present.

cam156 commented on spec/factories/file_sets.rb:
```diff
@@ -43,7 +43,7 @@
     trait :with_file_size do
       after(:build) do |fs|
-        fs.stub(:file_size).and_return '1234'
+        allow(fs).to receive(:file_size).and_return("1234")
```
@awead should still be single quote instead of double quote

awead replied:
No, it's really a style point here.

awead replied:
ok, re-read this. I'm not intending to be argumentative, I just thought this was a question and not a request. This is only occurrence of a single quote in the file. It's double everywhere else. Could we move the discussion of single vs. double to our style guide and look at using Rubocop to enforce this?

cam156 commented on spec/presenters/work_show_presenter_spec.rb:
```diff
@@ -20,7 +20,7 @@
 
     context "with two files in the work" do
-      let(:solr_doc)    { SolrDocument.new(work.to_solr).merge!('member_ids_ssim' => ["thing1", "thing2"]) }
+      let(:solr_doc)    { SolrDocument.new(work.to_solr).to_h.merge!('member_ids_ssim' => ["thing1", "thing2"]) }
```
@awead I thought we were changing to respond with a hash.  Why is the to_h needed?  Maybe this should be replaced with a name solr_hash of the change is needed

awead replied:
There were two deprecation warnings related to Blacklight::Document. One of them was for the `to_hash` change, and the other was for this change. This warning specifically said to do this.

nicktiberi committed:
Update ESLint rules to allow flat ternary expressions (#993)

awead committed:
Grant ExternalApp admin permissions

ExternalApp behaves very similar to User, and with upcoming changes to
the API, we can use policies to govern what external applications are
allowed to do. For now, they have admin status and can effectively do
anything. But, making that explicit required a few additions to the
class and refactoring/updating the existing policies to ensure that
those permissions we honored.

awead committed:
Find files' S3 etags via Graphql

In order to support checksum verification of files migrated from
Scholarsphere 3, we need to get file metadata information via the API.

This approach uses our graphql endpoint to query for a file using its
NOID from Scholarsphere 3. Because there it not a reliable way to
authorize the file via our Pundit policies, the endpoint query is only
available to administrators.

A user may post a graphql query and request a file's etag from Amazon,
which can be used to verify the file's integrity by calculating an
identical etag from the source in Scholarsphere 3.

nicktiberi committed:
Creators search UX improvements: (#996)

- hitting the Enter key in the creators search field no longer submits the form
- the first search result is automatically selected

awead committed:
Merge pull request #999 from psu-stewardship/graphql-etag

From Release 4.6.1

nicktiberi committed:
Action Footer UX Improvements: (#1000)

- reverse visual order of 'Save and Continue' and 'Cancel' buttons
- default form action is now 'Save and Continue' rather than 'Save as Draft'
- improve mobile/small device rendering of action footer buttons

awead committed:
Update files with metadata from Fits (#998)

Adds a :fits parameter to the files API endpoint used by our
metadata-listener application. It can now post metadata extracted from
Fits and have it persisted to the file's metadata field.

Note: Any parameters from the fits hash are permitted. When posting
updated information, the entire hash is replaced. There are no
differential updates to Fits's data. Virus, and other metadata
*outside* the fits hash key is retained.

nicktiberi committed:
Autopopulate display name when user adds a custom creator (#1003)

In the creators search, if the user selects one of the options to
create their own creator, prepopulate the custom creator's display
name with the content of the user's search query.

dependabot[bot] committed:
Bump rexml from 3.2.4 to 3.2.5 (#1002)

Bumps [rexml](https://github.com/ruby/rexml) from 3.2.4 to 3.2.5.
- [Release notes](https://github.com/ruby/rexml/releases)
- [Changelog](https://github.com/ruby/rexml/blob/master/NEWS.md)
- [Commits](https://github.com/ruby/rexml/compare/v3.2.4...v3.2.5)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
